Sources/Bot/Extensions/JSONDecoderExtensions.swift

```diff
@@ -0,0 +1,17 @@
+import Foundation
+import Result
+
+extension JSONDecoder {
+
+    static func with(dateDecodingStrategy strategy: DateDecodingStrategy) -> JSONDecoder {
+        let decoder = JSONDecoder()
+        decoder.dateDecodingStrategy = strategy
+        return decoder
+    }
+
+    func decode<T: Decodable>(_ payload: Data) -> Result<T, DecodingError> {
+        return Result(catching: { () -> T in
+            try decode(T.self, from: payload)
+        })
```

Comment:
[nit] trailing closure syntax would seem fit to me here, but 🤷‍♂ 

```swift
        return Result {
            try decode(T.self, from: payload)
        }
```

Reply:
That leads to an ambiguous compile error between `init(attempt:)` and `init(catching:)`.

Reply:
Oh, the deprecated one, right? Damn it 😞 Can't wait until we migrate to Swift 5 and use the official one 😅